hbase-server/src/main/java/org/apache/hadoop/hbase/master/normalizer/RegionNormalizerWorker.java

```diff
@@ -73,6 +78,8 @@ class RegionNormalizerWorker implements PropagatingConfigurationObserver, Runnab
     this.mergePlanCount = 0;
     this.rateLimiter = loadRateLimiter(configuration);
     this.defaultNormalizerTableLevel = extractDefaultNormalizerValue(configuration);
+    this.cumulativePlansSizeLimitMb =
+      configuration.getLong(CUMULATIVE_SIZE_LIMIT_MB_KEY, DEFAULT_CUMULATIVE_SIZE_LIMIT_MB);
```

Comment:
make sure to give this a similar treatment to the rateLimiter -- it needs to get parsed in onConfigurationChange below so that the limit can be updated dynamically. We should add a log there as well (formatted similar to the rate limit one) to inform the user that the limit has been updated.

when making that change, i know you dont see it in the other normalizer-related fields but typically i notice that most things updated by onConfigurationChange end up being volatile. Might be good to make cumulativePlansSizeLimitMb volatile along those lines.

Reply:
Added logging on change. Made cumulativePlansSizeLimitMb an AtomicLong because assignments to volatile longs aren't atomic.